tjsendy/tjsendy.php

```diff
@@ -0,0 +1,71 @@
+<?php
+/**
+ * @package    Tjsendy
+ * @author     Techjoomla <extensions@techjoomla.com>
+ * @copyright  Copyright (c) 2009-2020 TechJoomla. All rights reserved.
+ * @license    GNU General Public License version 2 or later.
+ */
+
+defined('_JEXEC') or die();
+
+use Joomla\CMS\Http\HttpFactory;
+use Joomla\CMS\Factory;
+
+jimport('joomla.event.plugin');
+
+/**
+ * Plug-in to save subscriber to sendy list
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class plgRDMediaTjsendy extends JPlugin
+{
+
+	protected $autoloadLanguage = true;
+
+	public function onAfterProcessSubscription($data)
+	{
+		$productCat = $this->getCategory($data['product_id']);
+		$user = Factory::getUser($data['userid']);
+
+		foreach ($this->params['field-name'] as $value)
+		{
+			$lists[] = $value;
+		}
+
+		foreach ($lists as $list)
+		{
+			$array[$list->category] = $list->listid;
+		}
+
+		$this->subscribeUser($user,$array[$productCat]);
+	}
+
+	function getCategory($productId)
+	{
+		$db = Factory::getDbo();
+
+		$query = $db->getQuery(true)
+			->select('category_id')
+			->from($db->quoteName('#__rd_subs_product2category'))
+			->where($db->quoteName('product_id') . ' = ' . $db->quote($productId));
```

Comment:
Numeric fields don't need to be quoted. You should typecast as `int` and not quote.